--- a/drivers/net/wireless/ath/regd.c
+++ b/drivers/net/wireless/ath/regd.c
@@ -638,8 +638,13 @@ ath_regd_init_wiphy(struct ath_regulatory *reg,
 	const struct ieee80211_regdomain *regd;
 
 	wiphy->reg_notifier = reg_notifier;
-	wiphy->regulatory_flags |= REGULATORY_STRICT_REG |
-				   REGULATORY_CUSTOM_REG;
+
+	if (IS_ENABLED(CONFIG_ATH_REG_DYNAMIC_USER_CERT_TESTING)) {
+		wiphy->regulatory_flags |= REGULATORY_CUSTOM_REG;
+	} else {
+		wiphy->regulatory_flags |= REGULATORY_STRICT_REG |
+					   REGULATORY_CUSTOM_REG;
+	}
 
 	if (ath_is_world_regd(reg)) {
 		/*
